lib/kitchen/directions/eoc_section_title_link_snippet.rb

```diff
@@ -4,30 +4,31 @@ module Kitchen
   module Directions
     module EocSectionTitleLinkSnippet
       def self.v1(page:)
-        chapter = page.ancestor(:chapter)
         <<~HTML
           <a href="##{page.title.id}">
-            <h3 data-type="document-title" id="#{page.title.copied_id}">
-              <span class="os-number">#{chapter.count_in(:book)}.#{page.count_in(:chapter)}</span>
-              <span class="os-divider"> </span>
-              <span class="os-text" data-type="" itemprop="">#{page.title_text}</span>
-            </h3>
+            #{title_snipet(page: page)}
           </a>
         HTML
       end
 
       def self.v2(page:)
-        chapter = page.ancestor(:chapter)
         <<~HTML
           <div>
-            <h3 data-type="document-title" id="#{page.title.copied_id}">
-              <span class="os-number">#{chapter.count_in(:book)}.#{page.count_in(:chapter)}</span>
-              <span class="os-divider"> </span>
-              <span class="os-text" data-type="" itemprop="">#{page.title_text}</span>
-            </h3>
+            #{title_snipet(page: page)}
           </div>
         HTML
       end
+
+      def self.title_snipet(page:, title_tag: 'h3')
+        chapter = page.ancestor(:chapter)
+        <<~HTML
+          <#{title_tag} data-type="document-title" id="#{page.title.copied_id}">
+            <span class="os-number">#{chapter.count_in(:book)}.#{page.count_in(:chapter)}</span>
+            <span class="os-divider"> </span>
+            <span class="os-text" data-type="" itemprop="">#{page.title_text}</span>
+          </#{title_tag}>
+        HTML
+      end
```

Comment:
we discussed this & i think this could be 1 function with a couple of different specification options!